models/user/index.js

```diff
@@ -414,6 +414,37 @@ class User extends Model {
     return output;
   }
 
+
+    /**
+   * Find [limit] new users (newlycreated and veryfied)
+   * who has at least [similarTagsNumber] similar tags to mine
+   * @param {int, int} limit, similatTagsNumber
+   * @returns {Promise<Object>}
+   */
+  static async findNewUsersWithMyTags(myUsername, limit, commonTagsNumber) {
+    const query = `
+    FOR similarUser in (
+    FOR u IN users
+    FILTER u.username == @myUsername
+    FILTER TO_BOOL(u.email) == true
+    FOR v,e,p IN 2 OUTBOUND u
+      ANY userTag
+      COLLECT foundUser = v AGGREGATE numberOfSimilarTags = COUNT(p[1]) INTO u2 
+      RETURN {foundUser, numberOfSimilarTags }
+    )
+
+    
+SORT similarUser.foundUser.created DESC
+FILTER similarUser.numberOfSimilarTags >= @similarTagsNumber
+FILTER TO_BOOL(similarUser.foundUser.email) == true
+LIMIT @limit
+RETURN {username: similarUser.foundUser.username, commonTags: similarUser.numberOfSimilarTags}
```

Comment:
let's return the user, not just username. the data like `givenName` etc will be needed for displaying.